java/client/src/main/java/glide/api/models/GlideString.java

```diff
@@ -0,0 +1,100 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models;
+
+import java.util.Arrays;
+import lombok.Getter;
+
+// TODO docs for the god of docs
+public class GlideString {
+
+    @Getter private byte[] bytes;
+    private String string = null;
+
+    /** Flag whether possibility to convert to string was checked. */
+    private boolean conversionChecked = false;
+
+    private GlideString() {}
+
+    public static GlideString of(String string) {
+        var res = new GlideString();
+        res.string = string;
+        res.bytes = string.getBytes();
+        return res;
+    }
+
+    public static GlideString of(byte[] bytes) {
+        var res = new GlideString();
+        res.bytes = bytes;
+        return res;
+    }
+
+    @Override
+    public String toString() {
+        return getString();
+    }
+
+    public String getString() {
+        if (string != null) {
+            return string;
+        }
+
+        assert canConvertToString() : "Value cannot be represented as a string";
+        return string;
+    }
+
+    public synchronized boolean canConvertToString() {
```

Comment:
Use double-checked locking with an AtomicBoolean to reduce the overhead in calling this. conversionChecked should be an AtomicBoolean.

if (!conversionChecked) {
  synchronized (this) {
    if (!conversionChecked) {
        do the conversion etc.
        conversionChecked.set(true);
   }
  }
  etc

Reply:
Thank you, fixed